### PR TITLE
[3890] Update the Bulk Upload function to support larger amounts of files and be more precise

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/3890

### CODE REVIEW

File: ui/app/src/components/BulkUpload.tsx [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+  )
+}
+
+export default function BulkUpload(props: any) {
+  const { formatMessage } = useIntl();
+  const classes = useStyles({});
+  const {
+    onClose = () => {
+    },
+    path = '/static-assets/test',
```
Comment (rart) [inline ui/app/src/components/BulkUpload.tsx]: Fix

File: ui/app/src/components/BulkUpload.tsx [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+    site = 'editorial',
```
Comment (rart) [inline ui/app/src/components/BulkUpload.tsx]: Fix

File: ui/app/src/components/BulkUpload.tsx [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+    limit = 2
```
Comment (rart) [inline ui/app/src/components/BulkUpload.tsx]: Rename to maxSimultaneousUploads, default to 1

File: ui/app/src/components/BulkUpload.tsx [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+  } = props;
+  const [status, setStatus] = useState(null);
+
+  const onComplete = () => {
+    setStatus('complete');
+  };
+
+  const onUpload = () => {
+    setStatus('uploading');
+  };
```
Comment (rart) [inline ui/app/src/components/BulkUpload.tsx]: Think it makes sense to have a single prop for this: `onStatusChange`

File: ui/app/src/components/BulkUpload.tsx [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+}
+
+function RenderArrowSvg() {
+  return (
+    <svg aria-hidden="true" focusable="false" className="UppyIcon uppy-DragDrop-arrow" width="16" height="16"
+         viewBox="0 0 16 16">
+      <path d="M11 10V0H5v10H2l6 6 6-6h-3zm0 0" fillRule="evenodd"/>
+    </svg>
+  )
+}
```
Comment (rart) [inline ui/app/src/components/BulkUpload.tsx]: - Format nicely
- Rename to ArrowPointingDown

File: ui/app/src/components/BulkUpload.tsx [hunk 6]
```diff
@@ -0,0 +1,471 @@
...
+          endpoint: getBulkUploadUrl(site, path),
+          formData: true,
+          fieldName: 'file',
+          limit
+        })
+        .use(ProgressBar, {
+          target: generalProgress.current,
+          hideAfterFinish: false
+        })
+        .setMeta({ site });
```
Comment (rart) [inline ui/app/src/components/BulkUpload.tsx]: Format

File: ui/app/src/components/DialogTitle.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
+/*
```
Comment (rart) [inline ui/app/src/components/DialogTitle.tsx]: All the annotations done on the Site Translation / Pages widgets PR for this component apply here.

File: ui/app/src/utils/string.ts [hunk 8]
```diff
@@ -40,6 +40,14 @@ export function isBlank(str: string): boolean {
   return str === '';
 }
 
+export function bytesToSize(bytes: number, seperator: string = '') {
```
Comment (rart) [inline ui/app/src/utils/string.ts]:
```ts
export function bytesToSize(bytes: string, separator: string = '');
export function bytesToSize(bytes: number, separator: string = '');
export function bytesToSize(bytes: number | string, separator: string = '') {
  ...
```
(notice typo in the word separator)

Add to default exports

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/BulkUpload.tsx (hunk 5 above)
Comment (aachavarria) [reply ui/app/src/components/BulkUpload.tsx]: removed

File: ui/app/src/components/DialogTitle.tsx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+}));
+
+export interface DialogTitleProps {
+  title: string;
+  subtitle?: string;
+  icon?: any;
+  onClose(): void;
+}
+
+export default function DialogTitle(props: DialogTitleProps) {
```
Comment (aachavarria) [inline ui/app/src/components/DialogTitle.tsx]: Change the name to DialogHeader

File: ui/app/src/components/DialogTitle.tsx [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+  const classes = dialogTitleStyles({});
+  const { onClose, title, subtitle, icon: Icon = CloseIcon } = props;
+  return (
+    <MuiDialogTitle disableTypography className={classes.titleRoot}>
+      <div className={classes.title}>
+        <Typography variant="h6">{title}</Typography>
```
Comment (aachavarria) [inline ui/app/src/components/DialogTitle.tsx]: add props (with default values) for titleTypographyProps, subtitleTypographyProps. Put variants in there

**Review by rart — APPROVED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**